src/core/net/udp6.cpp

```diff
@@ -229,10 +229,11 @@ Error Udp::Bind(SocketHandle &aSocket, const SockAddr &aSockAddr, otNetifIdentif
         do
         {
             aSocket.mSockName.mPort = GetEphemeralPort();
+        } while (aSocket.mSockName.mPort == Tmf::kUdpPort
```

Comment:
nit: Should we move this logic to `GetEphemeralPort` so that it never returns `Tmf::kUdpPort` ?

Reply:
SGTM!